src/main/java/frc/robot/subsystems/BuddyBalanceSubsystem.java

```diff
@@ -92,19 +104,52 @@ public void deployBuddyBalance() {
   }
 
   public void retrieveBuddy() { // Used to pick up the buddy robot while the lift is already underneath it
-    rightPIDController.setReference(kBalancedPosition, CANSparkMax.ControlType.kPosition);
-    leftPIDController.setReference(kBalancedPosition, CANSparkMax.ControlType.kPosition);
+    rightPIDController.setSetpoint(kRetrievedAngle);
+    leftPIDController.setSetpoint(kRetrievedAngle);
+    //buddyBalancePosEntry.setString("Raised");
   }
 
   public void releaseBuddy() { // Used to set down the robot
-    rightPIDController.setReference(kDeployedPosition, CANSparkMax.ControlType.kPosition);
-    leftPIDController.setReference(kDeployedPosition, CANSparkMax.ControlType.kPosition);
+    rightPIDController.setSetpoint(kDeployedAngle);
+    leftPIDController.setSetpoint(kDeployedAngle);
+    //buddyBalancePosEntry.setString("Lowered");
+  }
+
+  public void updateMotors() {
+    rightMotor.set(MathUtil.clamp(rightPIDControllerEncoderAngle, -1, 1));
+    leftMotor.set(MathUtil.clamp(leftPIDControllerEncoderAngle, -1, 1));
+  }
+
+  // These 4 methods are used for JUnit testing only
+  public void close() throws Exception {
+    // This method will close all device handles used by this object and release any other dynamic memory.
+    // Mostly for JUnit tests
+    rightMotor.close();
+    leftMotor.close();
+    activateDeploy.close();
+    rightEncoder.close();
+    leftEncoder.close();
+  }
+
+  public double getServoAngle() {
+    return activateDeploy.getAngle();
+  }
+
+  public double getEncoderRightAngle() {
+    return rightEncoder.getAbsolutePosition();
+  }
+
+  public double getEncoderLeftAngle() {
+    return leftEncoder.getAbsolutePosition();
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
-    buddyBalancePosition.append(rightEncoder.getPosition()); // Logging the position of the buddy balance lift
-    buddyBalancePosition.append(leftEncoder.getPosition());
+    updateMotors();
```

Comment:
Move this to the commands that you are running

Reply:
Run this as a default RepeatCommand, don't run it in periodic

Reply:
default repeat command didn't work in testing.